py/mpconfig.h

```diff
@@ -43,6 +43,9 @@
     MP_STRINGIFY(MICROPY_VERSION_MINOR) "." \
     MP_STRINGIFY(MICROPY_VERSION_MICRO)
 
+// String used for the banner, and sys.version additional information
+#define MICROPY_BANNER_VERSION "MicroPython " MICROPY_GIT_TAG " on " MICROPY_BUILD_DATE
```

Comment:
Could we add an optional prefix here for derivatives? For example we call our derivative "Pybricks MicroPython". On LEGO MINDSTORMS EV3, it can run along side vanilla MicroPython, so it is useful to have an easy way to see which one is running.

Reply:
Do you already have a banner that says `Pybricks MicroPython ...`?  If so, I guess it doesn't hurt to accommodate that, eg via:
```c
#ifndef MICROPY_BANNER_PREFIX
#define MICROPY_BANNER_PREFIX "MicroPython "
#endif

#define MICROPY_BANNER_VERSION MICROPY_BANNER_PREFIX MICROPY_GIT_TAG ...
```

Reply:
Yes, we did that in one release already (we had a patch that we kept rebasing on top of MicroPython).

Reply:
I made it so this entire macro is configurable by a board, which is more general that being able to add a prefix.  Does that work for you?